src/shims/unix/freebsd/foreign_items.rs

```diff
@@ -34,11 +34,14 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             "pthread_get_name_np" => {
                 let [thread, name, len] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
-                // FreeBSD's pthread_get_name_np does not return anything.
+                // FreeBSD's pthread_get_name_np does not return anything
+                // and uses  strlcpy, which truncates the resulting value,
+                // but always null terminates (except for zero sized buffers).
```

Comment:
```suggestion
                // and uses strlcpy, which truncates the resulting value,
                // but always adds a null terminator (except for zero-sized buffers).
```
Also, do you have source for this? Ideally some documentation, since implementation details can change.